Example/Tests/GINISupportedFormatsViewControllerTests.swift

```diff
@@ -0,0 +1,96 @@
+        
+        XCTAssertEqual(section0ItemsCount, tableSection0ItemsCount, "items count inside section 0 and table section 0 items count should be always equal")
+    }
+    
+    func testTableCellText() {
+        let indexPath = IndexPath(row: 0, section: 0)
+        let textForItem0AtSection0 = supportedFormatsViewController.sections[indexPath.section].items[indexPath.row]
+        
+        let textForCellAtIndexPath = supportedFormatsViewController.tableView(supportedFormatsViewController.tableView, cellForRowAt: indexPath).textLabel?.text
+        
+        XCTAssertEqual(textForCellAtIndexPath, textForItem0AtSection0, "text for item 0 at section 0 should be equal to the one declared on initialization")
+    }
+    
+    func testSectionHeaderHeight() {
+        let sectionHeaderHeight = supportedFormatsViewController.sectionHeight
+        
+        let tableSectionHeaderHeight = supportedFormatsViewController.tableView.sectionHeaderHeight
+        
+        XCTAssertEqual(sectionHeaderHeight, tableSectionHeaderHeight, "table view section header height should be equal to the one declare on initialization")
+    }
+    
+    func testRowHeight() {
+        let rowHeight = supportedFormatsViewController.rowHeight
+        
+        let tableRowHeight = supportedFormatsViewController.tableView.rowHeight
+        
+        XCTAssertEqual(rowHeight, tableRowHeight, "table view row height should be equal to the one declare on initialization")
+    }
+    
+    func testSectionTitle() {
+        let section0Title = supportedFormatsViewController.sections[0].title
+        
+        let tableSection0Title = supportedFormatsViewController.tableView(supportedFormatsViewController.tableView, titleForHeaderInSection: 0)
+        
+        XCTAssertEqual(section0Title, tableSection0Title, "table view section 0 title should be equal to the one declare on initialization")
+    }
+    
+    func testRowSelectionDisabled() {
+        let selectionState = supportedFormatsViewController.tableView.allowsSelection
+        
+        XCTAssertFalse(selectionState, "table view cell selection should not be allowed")
+    }
+    
+    func testSectionImageItemBackgroundColor() {
+        let indexPath = IndexPath(row: 0, section: 0)
+
+        let sectionImageItemBackgroundColor = supportedFormatsViewController.sections[indexPath.section].itemsImageBackgroundColor
+        
+        let cell = supportedFormatsViewController.tableView.cellForRow(at: indexPath) as! SupportedFormatsTableViewCell
+        let cellImageBackgroundColor = cell.imageBackgroundView.backgroundColor
+        
+        XCTAssertEqual(sectionImageItemBackgroundColor, cellImageBackgroundColor, "cell iage background color should be the same as the one declared on initialization")
+    }
+    
+    func testTableViewCellType() {
+        let indexPath = IndexPath(row: 0, section: 0)
+
+        let cell = supportedFormatsViewController.tableView.cellForRow(at: indexPath)
```

Comment:
You are always testing the cell at (0, 0). You should consider writing a getter method for it so the code is not duplicated in all tests.

Additionally, it might be helpful to also test different cells. Even if they are the same, checking different ones will ensure the same style is applied to all of them.

Reply:
I added some mocked sections so I can test them too, not only the ones embedded in the view controller.

I don't think that is really necessary to create a method just for that, I would leave as it is.

Reply:
Still not convinced. Now the only thing you test is `SupportedFormatsViewController `'s ability to create cells based on the `sections` array.

But you also need to test that it creates the cells specifically needed for the app at runtime. If the specs say "We need three cells with the following contents", you should also test that. What if one cell goes missing? What if the sections array is not set properly at runtime and no cells are shown?